### PR TITLE
add test cases to show empty string unmarshal results

### PR DESCRIPTION
**Pull request by danny-goldberg-transcarent**
created to show:
- https://github.com/rickb777/date/issues/25

### CODE REVIEW

File: marshal_test.go [hunk 1]
```diff
@@ -153,6 +153,7 @@ func TestDate_UnmarshalText_invalid_date_text(t *testing.T) {
 		{`not-a-date`, "date.ParseISO: cannot parse \"not-a-date\": year has wrong length\nmonth has wrong length\nday has wrong length"},
 		{`foot-of-og`, "date.ParseISO: cannot parse \"foot-of-og\": invalid year\ninvalid month\ninvalid day"},
 		{`215-08-15`, `date.ParseISO: cannot parse "215-08-15": invalid year`},
+		{``, `date.ParseISO: cannot parse "": invalid year`},
```
Comment (danny-goldberg-transcarent) [inline marshal_test.go:156]: this test fails since empty string parses to the date zero value

File: marshal_test.go [hunk 2]
```diff
@@ -162,3 +163,43 @@ func TestDate_UnmarshalText_invalid_date_text(t *testing.T) {
 		}
 	}
 }
+
+func TestDate_UnmarshalText_empty(t *testing.T) {
+	cases := []struct {
+		value    string
+		expected Date
+	}{
+		{``, New(1, time.January, 1)},
```
Comment (danny-goldberg-transcarent) [inline marshal_test.go:172]: this passes, proving that empty string parses as the date zero value - I believe that in date v1 the zero value was epoch, which is where we saw this initially

File: marshal_test.go [hunk 3]
```diff
@@ -162,3 +163,43 @@ func TestDate_UnmarshalText_invalid_date_text(t *testing.T) {
...
+
+func TestDate_UnmarshalTime_invalid_time_text(t *testing.T) {
+	cases := []struct {
+		value string
+		want  string
+	}{
+		{`not-a-date`, "parsing time \"not-a-date\" as \"2006-01-02T15:04:05Z07:00\": cannot parse \"not-a-date\" as \"2006\""},
+		{`foot-of-og`, "parsing time \"foot-of-og\" as \"2006-01-02T15:04:05Z07:00\": cannot parse \"foot-of-og\" as \"2006\""},
+		{`215-08-15`, `parsing time "215-08-15" as "2006-01-02T15:04:05Z07:00": cannot parse "215-08-15" as "2006"`},
+		{``, `parsing time "" as "2006-01-02T15:04:05Z07:00": cannot parse "" as "2006"`},
```
Comment (danny-goldberg-transcarent) [inline marshal_test.go:193]: these all pass, as `time.Time` unmarshal fails in the same ways (except in the empty string case)

File: marshal_test.go (hunk 2 above)
Comment (rickb777) [reply marshal_test.go:172]: Both v1 and v2 have the same UnmarshalText behaviour. The caveat is that in v1, the zero value was 1st Jan 1970; in v2 this has moved back to 1st Jan 1AD (the same as time.Time uses).

`UnmarshalText` states "... that the a blank string is unmarshalled as the zero value." Are you hoping for this to change?

File: marshal_test.go (hunk 3 above)
Comment (rickb777) [reply marshal_test.go:193]: Thanks for this contribution - there was a gap in the tests which you've filled in.

File: marshal_test.go (hunk 2 above)
Comment (danny-goldberg-transcarent) [reply marshal_test.go:172]: Thanks for the quick review!

I was hoping for the behavior to change. I had found it surprising that a `date.Date` in a struct would convert an empty string to a zero value date using the built-in unmarshal function (rather than failing as it would for `time.Time`)

~Also apologies that one of the tests I added does fail - I had added it to showcase what I believed was unintended behavior: https://github.com/rickb777/date/pull/24#discussion_r1691543911~

EDIT: nevermind, I see you built on this in https://github.com/rickb777/date/commit/79350eb05ce669566f3c96e8cb51ea0875ce7fc3 - thanks so much!
